bluebutton-data-model-codegen/src/main/java/gov/hhs/cms/bluebutton/data/model/codegen/MappingSpec.java

```diff
@@ -352,4 +358,104 @@ public String toString() {
 		builder.append("]");
 		return builder.toString();
 	}
+
+	/**
+	 * @return <code>true</code> if the RIF layout has an inner join relationship
+	 *         <code>false</code> if not
+	 */
+	public Boolean getHasInnerJoinRelationship() {
+		return this.hasInnerJoinRelationship;
+	}
+
+	/**
+	 * @param hasInnerJoinRelationship
+	 *            the new value for {@link #getHasInnerJoinRelationship()}
+	 * @return this {@link MappingSpec} instance, for call-chaining purposes
+	 */
+	public MappingSpec setHasInnerJoinRelationship(Boolean hasInnerJoinRelationship) {
+		this.hasInnerJoinRelationship = hasInnerJoinRelationship;
+		return this;
+	}
+
+	/**
+	 * @param innerJoinRelationship
+	 *            a list of {@link String} parameters defining the inner join
+	 *            relationship for the entity
+	 * @return this {@link MappingSpec} instance, for call-chaining purposes
+	 */
+	public MappingSpec setInnerJoinRelationship(List<List<String>> innerJoinRelationship) {
+		this.innerJoinRelationship = innerJoinRelationship;
+		return this;
+	}
+
+	/**
+	 * @return the list of {@link #innerJoinRelationship}s
+	 */
+	public List<List<String>> getInnerJoinRelationship() {
+		return this.innerJoinRelationship;
+	}
+
+	/**
+	 * @return the {@link ClassName} of the JPA {@link Entity} class that is a child
+	 *         of the header class
+	 */
+	public ClassName getChildEntity() {
+		return ClassName.get(packageName, childEntity);
+	}
+
+	/**
+	 * @param joinColumn
+	 *            the column to join on
+	 * @param parentEntity
+	 *            the parent entity of the current entity
+	 * @return this {@link MappingSpec} instance, for call-chaining purposes
+	 */
+	public MappingSpec setParentRelationship(String joinColumn, String parentEntity) {
+		this.joinColumn = joinColumn;
+		this.parentEntity = parentEntity;
+		return this;
+	}
+
+	/**
+	 * @return <code>true</code> if the RIF layout has a parent relationship
+	 *         <code>false</code> if not
+	 */
+	public boolean getHasParentRelationship() {
+		return hasParentRelationship;
+	}
```

Comment:
I don't know that you need a separate field for this? Can't it be computed based on whether or not `parentEntity != null && joinColumn != null`?